crates/solvers/src/domain/solver/baseline.rs

```diff
@@ -72,10 +72,21 @@ impl Baseline {
                             })
                             .collect();
 
+                        // The baseline solver generates a path with swapping
+                        // for exact output token amounts. This leads to
+                        // potential rounding errors for buy orders, where we
+                        // can buy slightly more than intended. Fix this by
+                        // capping the output amount to the order's buy amount
+                        // for buy orders.
+                        let mut output = route.output();
+                        if let order::Side::Buy = order.side {
+                            output.amount = cmp::min(output.amount, order.buy.amount);
+                        }
```

Comment:
It looks like these rounding issues can also cause us to buy slightly less than expected which also results in bad native prices. This can easily be reproduced on xdai.
Maybe we have to fix the rounding issue in the baseline algorithm itself?

Reply:
I think if we are confident that the algorithm is sound and these are indeed only rounding issues we could consider replacing `cmp::min` with something like this:
```rust
if matches!(order.side, order::Side::Buy) && is_roughly_equal(output.amount, order.buy.amount) {
    output.amount = order.buy.amount
}
```
This should be a last resort, though.